src/vs/workbench/contrib/debug/browser/debugANSIHandling.ts

```diff
@@ -96,23 +86,122 @@ export function handleANSIOutput(text: string, linkDetector: LinkDetector): HTML
 			} else {
 				currentPos = startPos;
 			}
-
 		}
 
 		if (sequenceFound === false) {
 			buffer += text.charAt(currentPos);
 			currentPos++;
 		}
-
 	}
 
 	// Flush remaining text buffer if not empty.
 	if (buffer) {
-		appendStylizedStringToContainer(root, buffer, styleNames, linkDetector);
+		appendStylizedStringToContainer(root, buffer, styleNames, linkDetector, customFgColor, customBgColor);
 	}
 
 	return root;
 
+	/**
+	 * Change the foreground or background color by clearing the current color
+	 * and adding the new one.
+	 * @param newClass If string or number, new class will be
+	 *  `code-(foreground or background)-newClass`. If `null`, no new class
+	 * 	will be added.
+	 * @param isForeground If `true`, will change the foreground color, if
+	 * 	`false`, will change the background color.
+	 * @param customColor If provided, this custom color will be used instead of
+	 * 	a class-defined color.
+	 */
+	function changeColor(newClass: string | number | null, isForeground: boolean, customColor?: RGBColor | null): void {
+		const colorType = isForeground ? 'foreground' : 'background';
+		styleNames = styleNames.filter(style => !style.match(new RegExp(`^code-${colorType}-(\\d+|custom)$`)));
+		if (newClass) {
+			styleNames.push(`code-${colorType}-${newClass}`);
+		}
+		if (isForeground) {
+			customFgColor = customColor || null;
+		} else {
+			customBgColor = customColor || null;
+		}
+	}
+
+	/**
+	 * Calculate and set basic ANSI formatting. Supports bold, italic, underline,
+	 * normal foreground and background colors, and bright foreground and
+	 * background colors. Not to be used for codes containing advanced colors.
+	 * Will ignore invalid codes.
+	 * @param styleCodes Array of ANSI basic styling numbers, which will be
+	 * applied in order. New colors and backgrounds clear old ones; new formatting
+	 * does not.
+	 * @see {@link https://en.wikipedia.org/wiki/ANSI_escape_code }
+	 */
+	function calcAndSetBasicFormatters(styleCodes: number[]): void {
+		for (let code of styleCodes) {
+			if (code === 0) {
```

Comment:
How about using a `switch` statement here. Would make it more readable imho

Reply:
How would you prefer I implement that with the ranges? List each number (there's 38), or use a `switch(true)` statement with ranges?

Reply:
I guess the switch would just make things overly comples. Feel free to keep if else